ThreeDSecure/src/main/java/com/braintreepayments/api/ThreeDSecureClient.java

```diff
@@ -418,16 +419,29 @@ private void startVerificationFlow(FragmentActivity activity, Configuration conf
 
         // perform cardinal authentication
         braintreeClient.sendAnalyticsEvent("three-d-secure.verification-flow.started");
-        if (observer != null) {
-             observer.launch(result);
-        } else {
-            Bundle extras = new Bundle();
-            extras.putParcelable(ThreeDSecureActivity.EXTRA_THREE_D_SECURE_RESULT, result);
 
-            Intent intent = new Intent(activity, ThreeDSecureActivity.class);
-            intent.putExtras(extras);
+        try {
+            if (observer != null) {
+                observer.launch(result);
+            } else {
+                Bundle extras = new Bundle();
+                extras.putParcelable(ThreeDSecureActivity.EXTRA_THREE_D_SECURE_RESULT, result);
+
+                Intent intent = new Intent(activity, ThreeDSecureActivity.class);
+                intent.putExtras(extras);
 
-            activity.startActivityForResult(intent, THREE_D_SECURE);
+                activity.startActivityForResult(intent, THREE_D_SECURE);
+            }
+        } catch (RuntimeException runtimeException) {
+            Throwable exceptionCause = runtimeException.getCause();
+            if (exceptionCause instanceof TransactionTooLargeException) {
+                String errorMessage = "The 3D Secure response returned is too large to continue.";
```

Comment:
Thinking out loud here, but I wonder if there are details that can be included in this message that would allow us to pinpoint problematic banks and escalate them to Cardinal. Should we also include the support email here and direct merchants to reach out to us if they encounter this error?

Reply:
Interesting yeah do we have a similar support message format on iOS we could borrow from?

Reply:
> I wonder if there are details that can be included in this message that would allow us to pinpoint problematic banks and escalate them to Cardinal

Merchants may be in the best position to report the root cause to support. We could technically dig into the `ThreeDSecureResult` for more info but I wouldn't know where to start.

Reply:
Oh that was a question... we add "Please contact support." to some of our error message that signals an error worth reaching out to us about.

Reply:
Ok cool yeah I wanted to make sure I didn't create a new type of support message format, that's legit. There's one in this class I can borrow from I'll add.